src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -370,11 +370,19 @@ private static bool IsAttributeSbrp(MetadataReader reader, CustomAttribute attr)
 
             if (attributeType == SbrpAttributeType)
             {
-                BlobReader blobReader = reader.GetBlobReader(attr.Value);
-                string attributeValue = Encoding.UTF8.GetString(blobReader.ReadBytes(blobReader.Length));
-                attributeValue = Regex.Replace(attributeValue, @"\p{C}+", string.Empty);
-                return Regex.IsMatch(attributeValue, SbrpAttributeValuePattern);
+                var provider = new CustomAttributeTypeProvider(reader);
+                var decodedValue = attr.DecodeValue(provider);
+                try
+                {
+                    string attributeValue = decodedValue.FixedArguments[0].Value.ToString() + " " + decodedValue.FixedArguments[1].Value.ToString();
+                    return attributeValue == SbrpAttributeValue;
```

Comment:
Instead of artificially formatting the two values that needs to be matched between here and the `const`, I would suggest just comparing each value separately.

```suggestion
                    return decodedValue.FixedArguments[0].Value.ToString() == "source" && decodedValue.FixedArguments[1].Value.ToString() == "source-build-reference-packages";
```

Reply:
Addressed [here](https://github.com/dotnet/installer/pull/17605/files#diff-a7b97c2b7dda64aa8ea1612d096041a356ad4d9712bbefa294d51e2b6391086dR375).